sleep_block.py

```diff
@@ -18,6 +21,48 @@ class Sleep(Block):
 
     interval = TimeDeltaProperty(title='Interval', default={'seconds': 1})
 
+    def __init__(self):
+        super().__init__()
+        self._signals = []
+        self._signals_lock = Lock()
+
+    def configure(self, context):
+        super().configure(context)
+        self._load_signals = self.persistence.load('signals') or []
+
+    def start(self):
+        super().start()
+        self._schedule_persistence_sleeps()
+
+    def stop(self):
+        super().stop()
+        self._store_signals(_time())
+        self.persistence.save()
```

Comment:
We obviously want to persist.save on stop, but we should also periodically do it throughout I think, just in case of something bad happening (crash, doesn't stop fast enough, etc). However, I don't think we need to add it to this block, I think it's probably looking like it's time for a persistence mixin...